### PR TITLE
added tests to `$kn service update ...` with --requests/limits-cpu/memory and removed required --image

### PR DESCRIPTION
**Pull request by maximilien**
* added tests for service update with variations of --requests/limits-cpu --requests/limits-memory
* made image optional on service update since that was not the case and image should be optional

### CODE REVIEW

**Comment by sixolet** [comment]
/approve
Change the top line of the commit message to something describing the --image requirement and I'll even lgtm.

**Comment by maximilien** [comment]
NOTE: making some refactoring changes to simplify tests
/hold-off

File: pkg/kn/commands/configuration_edit_flags.go [hunk 1]
```diff
@@ -45,6 +45,10 @@ func (p *ConfigurationEditFlags) AddFlags(command *cobra.Command) {
 	command.Flags().StringVar(&p.RequestsFlags.Memory, "requests-memory", "", "The requested CPU (e.g., 64Mi).")
 	command.Flags().StringVar(&p.LimitsFlags.CPU, "limits-cpu", "", "The limits on the requested CPU (e.g., 1000m).")
 	command.Flags().StringVar(&p.LimitsFlags.Memory, "limits-memory", "", "The limits on the requested CPU (e.g., 1024Mi).")
+}
+
+func (p *ConfigurationEditFlags) AddCreateFlags(command *cobra.Command) {
+	p.AddUpdateFlags(command)
```
Comment (navidshaikh) [inline pkg/kn/commands/configuration_edit_flags.go]: :+1:

**Comment by maximilien** [comment]
Looks like Knative setup is failing in integration tests... see log:

```
I0429 22:50:27.461] error: unable to read URL "https://storage.googleapis.com/knative-nightly/serving/latest/istio-crds.yaml", server reported 404 Not Found, status code=404
I0429 22:50:27.466] ERROR: Knative setup failed
[...]
```

**Comment by maximilien** [comment]
```
I0429 22:50:27.461] error: unable to read URL "https://storage.googleapis.com/knative-nightly/serving/latest/istio-crds.yaml", server reported 404 Not Found, status code=404
I0429 22:50:27.466] ERROR: Knative setup failed
[...]
```

Could be intermitent issue with Istio CRD. Let me trigger test again.

/test pull-knative-client-integration-tests

**Comment by maximilien** [comment]
Woohoo, looks like we are green!

cc: @sixolet @cppforlife

File: pkg/kn/commands/service_update_test.go [hunk 2]
```diff
@@ -107,3 +107,153 @@ func TestServiceUpdateImage(t *testing.T) {
 		t.Fatalf("wrong image set: %v", conf.RevisionTemplate.Spec.Container.Image)
 	}
 }
+
+func TestServiceUpdateRequestsLimitsCPU(t *testing.T) {
+	service := createMockServiceWithResources(t, "250", "64Mi", "1000m", "1024Mi")
+
+	action, updated, _, err := fakeServiceUpdate(service, []string{
+		"service", "update", "foo", "--requests-cpu", "500m", "--limits-cpu", "1000m"})
```
Comment (cppforlife) [inline pkg/kn/commands/service_update_test.go]: did you want to have it be different than `1000m`?

Comment (maximilien) [reply pkg/kn/commands/service_update_test.go]: Well, wanted to verify case where the update does nothing...

**Pull request merged**